### PR TITLE
[converter] Supports gstreamer video format RGBP, BGRP

### PR DESCRIPTION
**Pull request by anyj0527**
- Channel First (NCHW) formats RGBP and BGRP are supported in gstreamer 1.20.0+
- Let tensor_converter handle this video_format.

Simple performance test in my machine shows
  2500 FPS (videoconvert) v.s. 1300 FPS (tensor_transform transpose)

```bash
videotestsrc ! video/x-raw,width=640,height=480,format=RGB ! \
    videoscale ! videoconvert !
    video/x-raw,format=RGBP,width=320,height=320 ! \
    tensor_converter ! tensor_sink
 -> 2500 FPS
v.s.
gst-launch-1.0 \
videotestsrc ! video/x-raw,width=640,height=480,format=RGB ! \
    videoscale ! videoconvert !
    video/x-raw,format=RGB,width=224,height=224 ! \
    tensor_converter ! \
    tensor_transform mode=transpose option=1:2:0:3 ! tensor_sink
 -> 1300 FPS
```

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: gst/nnstreamer/elements/gsttensor_converter_media_info_video.h [hunk 1]
```diff
@@ -25,9 +25,15 @@
 /**
  * @brief Caps string for supported video format
  */
+#if GST_CHECK_VERSION(1, 20, 0)
+#define VIDEO_CAPS_STR \
```
Comment (jaeyun-jung) [inline gst/nnstreamer/elements/gsttensor_converter_media_info_video.h:29]: Suggestion: separate video format macro
```
/**
 * @brief Supported video format.
 */
#if GST_CHECK_VERSION(1, 20, 0)
#define NNS_VIDEO_FORMAT "{ RGB, BGR, RGBx, BGRx, xRGB, xBGR, RGBA, BGRA, ARGB, ABGR, GRAY8, GRAY16_BE, GRAY16_LE, RGBP, BGRP }"
#else
#define NNS_VIDEO_FORMAT "{ RGB, BGR, RGBx, BGRx, xRGB, xBGR, RGBA, BGRA, ARGB, ABGR, GRAY8, GRAY16_BE, GRAY16_LE }"
#endif

/**
 * @brief Caps string for supported video format
 */
#define VIDEO_CAPS_STR \
    GST_VIDEO_CAPS_MAKE (NNS_VIDEO_FORMAT) \
    ", interlace-mode = (string) progressive"
```

File: gst/nnstreamer/elements/gsttensor_converter.c [hunk 2]
```diff
@@ -1994,19 +2026,49 @@ gst_tensor_converter_get_possible_media_caps (GstTensorConverter * self)
...
 
             if (config.rate_n >= 0 && config.rate_d > 0) {
               gst_structure_set (st, "framerate", GST_TYPE_FRACTION,
                   config.rate_n, config.rate_d, NULL);
             }
-          }
+
+            /* add new structure for NCHW formats */
+#if GST_CHECK_VERSION(1, 20, 0)
+              width = config.info.info[0].dimension[0];
```
Comment (jaeyun-jung) [inline gst/nnstreamer/elements/gsttensor_converter.c:2044]: indent?

File: gst/nnstreamer/elements/gsttensor_converter_media_info_video.h (hunk 1 above)
Comment (anyj0527) [reply gst/nnstreamer/elements/gsttensor_converter_media_info_video.h]: Looks good! I followed your suggestion. PTAL

File: gst/nnstreamer/elements/gsttensor_converter.c (hunk 2 above)
Comment (anyj0527) [reply gst/nnstreamer/elements/gsttensor_converter.c]: I fixed those indents, PTAL

File: gst/nnstreamer/elements/gsttensor_converter.c [hunk 3]
```diff
@@ -1503,17 +1513,25 @@ gst_tensor_converter_parse_video (GstTensorConverter * self,
...
+      config->info.info[0].type = _NNS_UINT8;
+      config->info.info[0].dimension[0] = width;
+      config->info.info[0].dimension[1] = height;
+      config->info.info[0].dimension[2] = 3;
       break;
+#endif
     default:
       GST_WARNING_OBJECT (self,
-          "The given video caps with format \"%s\" is not supported. Please use GRAY8, GRAY16_LE, GRAY16_BE, RGB, BGR, RGBx, BGRx, xRGB, xBGR, RGBA, BGRA, ARGB, or ABGR.\n",
+          "The given video caps with format \"%s\" is not supported. Please use " NNS_VIDEO_FORMAT "\n",
```
Comment (jaeyun-jung) [inline gst/nnstreamer/elements/gsttensor_converter.c:1530]: IMO "\n" at last log string is unnecessary :)

**Review by jaeyun-jung — APPROVED** [review]

**Review by niley7464 — APPROVED** [review]
LGTM 👍

**Pull request merged**